### PR TITLE
Create JudgeTeam class

### PR DESCRIPTION
**Pull request by lowellrex**
Connects #7506.

In preparation for [moving attorney membership in judge teams](https://github.com/department-of-veterans-affairs/caseflow/pull/7847) over to the organization model this PR creates a new `JudgeTeam` class.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -209,6 +209,10 @@ def current_case_assignments
     self.class.appeal_repository.load_user_case_assignments_from_vacols(css_id)
   end
 
+  def administrated_teams
```
Comment (lowellrex) [inline app/models/user.rb]: The name of this function seems a little clumsy to me, but I struggle to think of a better one.

File: app/models/organizations/judge_team.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    user.administrated_teams.select { |team| team.is_a?(JudgeTeam) }.first
+  end
+
+  def self.create_for_judge(user)
+    create!(name: user.css_id).tap do |org|
+      OrganizationsUser.make_user_admin(user, org)
+    end
+  end
+
+  def can_receive_task?(_task)
```
Comment (mdbenjam) [inline app/models/organizations/judge_team.rb]: Why is this false?

File: spec/models/judge_team_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    context "when user is admin of JudgeTeam" do
+      let!(:judge_team) { JudgeTeam.create_for_judge(user) }
+
+      it "should return judge team" do
+        expect(JudgeTeam.for_judge(user)).to eq(judge_team)
+      end
+    end
+
+    # Limitation of the current approach is that users are effectively limited to being the admin of a single JudgeTeam.
```
Comment (mdbenjam) [inline spec/models/judge_team_spec.rb]: I think this is fine

File: app/models/user.rb (hunk 1 above)
Comment (mdbenjam) [reply app/models/user.rb]: I think I would either name it: `administered_teams` or `teams_admin_for`

**Pull request merged**